src/cdomains/pentagon/boxes.ml

```diff
@@ -0,0 +1,88 @@
+open ZExt
+open Intv
+open Batteries
+open StringUtils
+
+(** Provides functions and types for collections of Intv. *)
+module Boxes  = 
+struct
+  type t = Intv.t list [@@deriving eq, ord]
+  include ZExtOps
+  let is_bot (i: t) = 
+    BatList.exists Intv.is_bot i
+
+  let is_top i =
+    BatList.for_all Intv.is_top i
+
+  let to_string (intervals: t) =
+    let string_of_interval i intv =
+      Printf.sprintf "%i->%s" i (Intv.to_string intv)
+    in
+    let bot_or_top =
+      if is_bot intervals then
+        StringUtils.bot_str
+      else if is_top intervals then
+        StringUtils.top_str
+      else 
+        " "
+        |> Printf.sprintf "(%s)"
+    in
+    Printf.sprintf "%s { %s }" bot_or_top (String.concat "; " (List.mapi string_of_interval intervals))
+
+  let equal boxes1 boxes2 =
+    BatList.for_all2 (Intv.equal) boxes1 boxes2
+
+  let leq i1 i2 =
+    BatList.for_all2 Intv.leq i1 i2
+
+  let join (i1: t) (i2: t) = 
+    BatList.map2 Intv.union i1 i2
+
+  let meet (i1: t) (i2: t) = 
+    BatList.map2 Intv.inter i1 i2
+
+  let widen (i1: t) (i2: t) = 
+    BatList.map2 Intv.widen i1 i2
+
+  let narrow (i1: t) (i2: t) = 
+    meet i1 i2
```

Comment:
Could you elaborate on why setting narrow to meet is ok here, i.e., why the domain does not have (quasi-)infinite (i.e. of length 2^64 or something) descending chains.